core/domain/action_decorators.py

```diff
@@ -0,0 +1,121 @@
+# coding: utf-8
+#
+# Copyright 2017 The Oppia Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Decorators to provide authorization accross the site."""
+
+from core.domain import rights_manager
+from core.domain import role_services
+import feconf
+
+
+def check_activity_accessible(self, activity_id, activity_type):
+    """Returns a boolean to signify whether given activity is accessible
+    by the user or not.
+    """
+    if activity_type == feconf.ACTIVITY_TYPE_EXPLORATION:
+        if activity_id in feconf.DISABLED_EXPLORATION_IDS:
+            self.render_template(
+                'pages/error/disabled_exploration.html',
+                iframe_restriction=None)
+            return
+
+        # This check is needed in order to show the correct page when a 404
```

Comment:
So where should it go?
It was a part of @require_playable decorator which will be replaced now. Should I shift this code in each method where the decorator is used?